querier_test.go

```diff
@@ -1691,6 +1692,118 @@ func BenchmarkQuerySeek(b *testing.B) {
 	}
 }
 
+func BenchmarkSetMatcher(b *testing.B) {
+	cases := []struct {
+		numBlocks                   int
+		numSeries                   int
+		numSamplesPerSeriesPerBlock int
+		setPattern                  string
+		regexPattern                string
+	}{
+		{
+			numBlocks:                   1,
+			numSeries:                   15,
+			numSamplesPerSeriesPerBlock: 10,
+			setPattern:                  "^(?:1|2|3)$",
+			regexPattern:                "1|2|3",
```

Comment:
Instead of checking this, we can check `setPattern` itself for before and after this PR and get the diff. This would give the actual impact of the changes in this PR.

Reply:
You mean delete `regexPattern`?

Reply:
You can keep both, but test it before and after this PR.

Reply:
Just for reference: https://godoc.org/golang.org/x/tools/cmd/benchcmp